test/Npgsql.Tests/Replication/CommonReplicationTests.cs

```diff
@@ -454,6 +454,43 @@ async Task<NpgsqlLogSequenceNumber> GetCommitLsn(string valueString)
 
         #endregion
 
+        #region BugTests
+
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/3534")]
+        public Task Bug3534()
+            => SafeReplicationTest(
```

Comment:
If I understand the test correctly, the idea is basically that we should *not* time out - with this particular timer configuration - while waiting for messages. If so, then it's probably simpler to just do Task.WhenAll on the Task representing the read (the Task returned by AsyncEnumerator.MoveNextAsync) and a Task.Delay task, and make sure the Task.Delay task completes first (and there's no exception on the first). Once that's done, just cancel the replication and finish the test.